### PR TITLE
AMB-1441-Construct-Endpoint-Key-in-proxy

### PR DESCRIPTION
**Pull request by LKFULLER1**

### CODE REVIEW

File: proxies/live/apiproxy/resources/jsc/CreateKVMKeyName.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-var ods = "REPC"
+var ods = context.getVariable("jwt.DecodeJWT.DecodeIDToken.claim.ods_code")
 var interactionId = context.getVariable("request.header.Interaction-ID")
```
Comment (JalfResi) [inline proxies/live/apiproxy/resources/jsc/CreateKVMKeyName.js:2]: This is wrong! Interaction IDs should be fixed (see ticket description)

File: tests/test_mock_receiver.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 @pytest.mark.debug
 @pytest.mark.integration
 @pytest.mark.user_restricted_separate_nhs_login
-@pytest.mark.nhsd_apim_authorization({"access": "patient", "level": "P9"})
+@pytest.mark.nhsd_apim_authorization({"access": "patient", "level": "P9", "login_form": {"username": "9912003071"}})
 def test_mock_receiver_patient_record_path(nhsd_apim_proxy_url, nhsd_apim_auth_headers):
     headers = {
               "Interaction-ID": interaction_id, "X-Request-ID": "60E0B220-8136-4CA5-AE46-1D97EF59D068",
```
Comment (JalfResi) [inline tests/test_mock_receiver.py:15]: Client should not be providing interaction-id

**Review by almas-liaqat — APPROVED** [review]

**Pull request merged**